### PR TITLE
add support for encapsulating Ethernet frame in MPLS

### PR DESCRIPTION
**Pull request by haozheng10**
This commit adds support for encapsulating Ethernet frame in MPLS with control word per rfc 4448. 
You can do something like:
MPLS(label=1)/EoMCW(seq=0x1234)/Ether(dst='00:01:00:11:11:11',src='00::02:00:22:22:22')/IP(dst='10.0.0.1', src='10.0.0.2', proto=252)
or
MPLS(label=1)/Ether(dst='00:01:00:11:11:11',src='00::02:00:22:22:22')/IP(dst='10.0.0.1', src='10.0.0.2', proto=252)

### CODE REVIEW

File: scapy/contrib/mpls.py [hunk 1]
```diff
@@ -16,30 +16,47 @@
...
+                    BitField("reserved", 0, 12),
+                    ShortField("seq", 0) ]
+    def guess_payload_class(self, payload):
+        if len(payload) >= 1:
+            return Ether
+        return Padding
+
 class MPLS(Packet):
-   name = "MPLS"
-   fields_desc =  [ BitField("label", 3, 20),
```
Comment (guedou) [inline scapy/contrib/mpls.py]: Please do not change code when not needed, it breaks the history.

File: scapy/contrib/mpls.py [hunk 2]
```diff
@@ -16,30 +16,47 @@
...
-           if not self.s:
-              return MPLS
-           ip_version = (orb(payload[0]) >> 4) & 0xF
-           if ip_version == 4:
-               return IP
-           elif ip_version == 6:
-               return IPv6
-       return Padding
+    def guess_payload_class(self, payload):
+        if len(payload) >= 1:
```
Comment (guedou) [inline scapy/contrib/mpls.py]: Same.

**Comment by guedou** [comment]
Thanks for your PR. Could you also add tests in `test/regression.uts`

File: scapy/contrib/mpls.py (hunk 1 above)
Comment (haozheng10) [reply scapy/contrib/mpls.py]: The indentation is weird, first indentation 3 spaces, subsequent 4 spaces.  My editor automatically fixed it.
Do you not want the indentation fixed?

File: scapy/contrib/mpls.py (hunk 2 above)
Comment (haozheng10) [reply scapy/contrib/mpls.py]: same

**Comment by haozheng10** [comment]
added related tests to test/regression.uts

**Comment by guedou** [comment]
@haozheng10 Could you revert your unnecessary changes? Sorry, we have been busy with the 2.4.0 release.

**Comment by haozheng10** [comment]
@guedou I reverted the indentation fixes per your request

**Pull request merged**